course4/homework-string/string.hh

```diff
@@ -1,10 +1,35 @@
 #pragma once
+#include <ostream>
+#include <memory>
 
 namespace course {
-
 class string
 {
-	string();
+public:
+    string();
+    ~string();
+    string(const char* chars);
+    string(const string& source);
+
+    string& operator=(const string& source);
+    bool operator ==(const string& a);
+    bool operator ==(const char* chars);
+    bool operator !=(const string& a);
+    char& operator[](int index) const;
+
+    bool empty() const;
+    int size() const;
+
+private:
+    std::unique_ptr<char[]> mCharArray;
+    int mLength;
+    int getLenghtOfChars(const char* chars);
+    void deepCopy(const char* chars, int lenght);
+    bool deepCompare(const char* chars, int length);
 };
 
-}
\ No newline at end of file
+std::stringstream& operator <<(std::stringstream& os, const course::string &s);
```

Comment:
wrong! This will only worj with stringstresam, but not with std::cout or std::fstream. Use ```std::ostream```, it's the base class.